src/features/devices/models/device-additions.ts

```diff
@@ -428,4 +429,65 @@ export const addToModel = (abstractSql: AbstractSqlModel) => {
 			],
 		],
 	});
+
+	abstractSql.tables['device'].fields.push({
+		fieldName: 'should be running-release',
+		dataType: 'ForeignKey',
+		required: false,
+		references: {
+			resourceName: 'release',
+			fieldName: 'id',
+		},
+		computed: [
+			'Case',
+			[
+				'When',
+				['Exists', ['ReferencedField', 'device', 'is pinned on-release']],
+				['ReferencedField', 'device', 'is pinned on-release'],
+			],
+			[
+				'Else',
+				[
+					'SelectQuery',
+					[
+						'Select',
+						[['ReferencedField', 'application', 'should be running-release']],
+					],
+					['From', ['Table', 'application']],
+					[
+						'Where',
+						[
+							'Equals',
+							['ReferencedField', 'application', 'id'],
+							['ReferencedField', 'device', 'belongs to-application'],
+						],
+					],
+				],
+			],
+		],
+	});
+
+	abstractSql.relationships = _.merge(abstractSql.relationships, {
```

Comment:
Would it make more sense to generate these via having it in the sbvr and then converting it to computed in the abstract sql model?

Reply:
I was also thinking about that while writing this...
Didn't try it primarily b/c we have never done that before for any other computed term on the balena model, but yes it totally makes sense and works :+1: